str_analysis/call_non_ref_pathogenic_motifs.py

```diff
@@ -0,0 +1,978 @@
+        locus_results_json (dict): Results will be added to this dictionary.
+    """
+    canonical_motif = compute_canonical_repeat_unit(motif)
+    offtarget_regions = OFFTARGET_REGIONS[args.genome_version].get(canonical_motif)
+    if offtarget_regions is None:
+        print(f"WARNING: off-target regions not available for {canonical_motif}")
+        return
+
+    read_count_with_offtargets = read_count
+    with pysam.Samfile(args.bam_or_cram_path, reference_filename=args.reference_fasta) as f:
+        for offtarget_region in offtarget_regions:
+            offtarget_chrom, offtarget_start, offtarget_end = parse_interval(offtarget_region)
+            sequences = (r.seq for r in f.fetch(offtarget_chrom, offtarget_start, offtarget_end) if r.mapq >= MIN_MAPQ)
+            c, t = count_repeat_in_sequences(
+                sequences,
+                motif,
+                min_occurrences=3,
+                min_fraction_bases_covered=MIN_FRACTION_OF_BASES_COVERED)
+
+            read_count_with_offtargets += c
+            if args.verbose:
+                print(f"{c} out of {t} reads contained {motif} in off-target region {offtarget_region}")
+
+    locus_results_json.update({
+        f"motif{motif_number}_read_count_with_offtargets": read_count_with_offtargets,
+        f"motif{motif_number}_normalized_read_count_with_offtargets":
+            read_count_with_offtargets * NORMALIZE_TO_COVERAGE / flank_coverage_mean if flank_coverage_mean > 0 else 0,
+    })
+
+
+def process_locus(locus_id, args):
+    """Compute results for a single locus and write them to a json file.
+
+    Args:
+        locus_id (str): locus id
+        args (object): command-line args from argparse
+    """
+    locus_coords_1based = LOCUS_INFO[locus_id]["LocusCoords_1based"][args.genome_version]
+    locus_chrom, start_1based, end_1based = parse_interval(locus_coords_1based)
+    locus_start_0based = start_1based - 1
+    locus_end = end_1based
+    use_offtarget_regions = args.use_offtarget_regions or LOCUS_INFO[locus_id]["UseOfftargetRegions"]
+
+    known_pathogenic_motifs = list(map(compute_canonical_repeat_unit, LOCUS_INFO[locus_id]["Motifs"]["PATHOGENIC"]))
+    known_benign_motifs = list(map(compute_canonical_repeat_unit, LOCUS_INFO[locus_id]["Motifs"]["BENIGN"]))
+
+    pathogenic_motif_size = len(known_pathogenic_motifs[0])
+
+    # process bam/cram
+    overlapping_sequences, left_flank_n_well_aligned_bases, right_flank_n_well_aligned_bases = process_reads_in_locus(
+        args.bam_or_cram_path, args.reference_fasta, locus_chrom, locus_start_0based, locus_end, pathogenic_motif_size)
+
+    locus_results_json = {}
+    left_flank_coverage = left_flank_n_well_aligned_bases / FLANK_SIZE
+    right_flank_coverage = right_flank_n_well_aligned_bases / FLANK_SIZE
+    locus_results_json.update({
+        "sample_id": args.sample_id,
+        "locus_id": locus_id,
+        "locus_coords": f"{locus_chrom}:{start_1based}-{end_1based}",
```

Comment:
The choice of 1-based vs. 0-based format seems to me pretty arbitrary (similar to the choice between json vs. yaml for the output file). I like 1-based because of IGV, and other users that copy coords into IGV will find it convenient (like users that prefer .json will be glad I picked json instead of yaml), but I don't know if this type of choice needs additional documentation.